integration/framework.go

```diff
@@ -0,0 +1,297 @@
+	out, err := exec.Command(path.Join(kubeBaseDir, "cluster", "kube-down.sh")).CombinedOutput()
+	if err != nil {
+		glog.Errorf("failed to tear down cluster - %q\n%s", err, out)
+		return fmt.Errorf("failed to tear down kube cluster - %q", err)
+	}
+
+	return nil
+}
+
+func getMasterIP() (string, error) {
+	out, err := exec.Command("gcloud", "compute", "instances", "list", "kubernetes-master", "--format=yaml").CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	glog.V(2).Infof("Output of gcloud compute instance list kubernetes-master:\n %s", out)
+	for _, line := range strings.Split(string(out), "\n") {
+		if strings.Contains(line, "natIP") {
+			externalIPLine := strings.Split(line, ":")
+			return strings.TrimSpace(externalIPLine[1]), nil
+		}
+	}
+
+	return "", fmt.Errorf("External IP not found for the master. 'gcloud compute instances list kubernetes-master' returned %s", out)
+}
+
+func downloadRelease(workDir, version string) error {
+	// Temporary download path.
+	downloadPath := path.Join(workDir, "kube")
+	// Format url.
+	downloadUrl := fmt.Sprintf(imageUrlTemplate, version)
+	glog.V(1).Infof("About to download kube release using url: %q", downloadUrl)
+
+	// Download kube code and store it in a temp dir.
+	if err := exec.Command("wget", downloadUrl, "-O", downloadPath).Run(); err != nil {
+		return fmt.Errorf("failed to wget kubernetes release @ %s - %q", downloadUrl, err)
+	}
+
+	// Un-tar kube release.
+	if err := exec.Command("tar", "-xf", downloadPath, "-C", workDir).Run(); err != nil {
+		return fmt.Errorf("failed to un-tar kubernetes release at %s - %q", downloadPath, err)
+	}
+	return nil
+}
+
+func getKubeClient() (*kube_client.Client, error) {
+	masterIP, err := getMasterIP()
+	if err != nil {
+		return nil, err
+	}
+	glog.V(1).Infof("Kubernetes master IP is %s", masterIP)
+
+	config := kube_client.Config{
+		Host:    "https://" + masterIP,
+		Version: "v1beta1",
+	}
+
+	auth, err := kube_clientauth.LoadFromFile(*authConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error loading auth - %q", err)
+	}
+	config, err = auth.MergeWithConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating client - %q", err)
+	}
+
+	kubeClient, err := kube_client.New(&config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating client - %q", err)
+	}
+
+	return kubeClient, nil
+}
+
+func validateCluster(baseDir string) bool {
+	out, err := runKubeClusterCommand(baseDir, "validate-cluster.sh")
+	if err != nil {
+		glog.V(1).Infof("cluster validation failed - %q\n %s", err, out)
+		return false
+	}
+	return true
+}
+
+func newKubeFramework(t *testing.T, version string) (kubeFramework, error) {
+	// All integration tests are large.
+	if testing.Short() {
+		t.Skip("Skipping framework test in short mode")
+	}
+
+	if !exists(*authConfig) {
+		return nil, fmt.Errorf("failed to find kube auth file: %s", *authConfig)
+	}
+
+	// Create a temp dir to store the kube release files.
+	tempDir := path.Join(*workDir, version)
```

Comment:
The intent is to reuse kube releases across runs. That is the reason for using the same tempdir.